saleor/cart/utils.py

```diff
@@ -65,20 +65,31 @@ def check_product_availability_and_warn(request, cart):
         remove_unavailable_variants(cart)
 
 
-def find_and_assign_anonymous_cart(request, queryset=Cart.objects.all()):
+def find_and_assign_anonymous_cart(queryset=Cart.objects.all()):
     """Assign cart from cookie to request user
     :type request: django.http.HttpRequest
     """
-    token = request.get_signed_cookie(Cart.COOKIE_NAME, default=None)
-    if not token:
-        return
-    cart = get_anonymous_cart_from_token(token=token, cart_queryset=queryset)
-    if cart is None:
-        return
-    cart.change_user(request.user)
-    carts_to_close = Cart.objects.open().filter(user=request.user)
-    carts_to_close = carts_to_close.exclude(token=token)
-    carts_to_close.update(status=Cart.CANCELED, last_status_change=now())
+    def get_cart(view):
+        @wraps(view)
+        def func(request, *args, **kwargs):
+            token = request.get_signed_cookie(Cart.COOKIE_NAME, default=None)
+            if not token:
+                return
+            cart = get_anonymous_cart_from_token(
+                token=token, cart_queryset=queryset)
+            if cart is None:
+                return
+            cart.change_user(request.user)
```

Comment:
How do we know the user is logged in by this point? We have yet to execute `view(...)`.